### DISCUSSION TITLE
Recommended way to save features of subscription plan?

### DISCUSSION THREAD

**Discussion by atinder** [pay-rails/pay#863] [Ideas]
Additional table or column. what do you recommend?

**Comment by excid3** [comment]
@atinder I've been just using a separate table. We've been considering adding it to the gem as well.

You can also just hardcode a hash as a constant since the plans aren't likely to change often.

**Comment by archonic** [comment]
I'd like to add my 👍  for a convention on storing products and plans. It would be nice to have enough data locally to display a form which can switch between products and plans.

**Comment by deanpcmad** [comment]
I can work on adding this as it was an annoyance for me that it wasn't already included. 

Would we want 2 models, such as Product and Price? As such a Product (e.g. "My SaaS Plan"), can have many prices (e.g. "$5 monthly" or "$50 yearly")

**Comment by excid3** [comment]
Unless you're selling multiple products, there's not a need for a Product model.

**Comment by deanpcmad** [comment]
True. Maybe just a Plan model then?

**Comment by excid3** [comment]
As far as Pay's concerned, probably. 

I would probably call it `Pay::Price` since we'll want to support one-time purchases and not just subscriptions.

Something like this:

#### `Pay::Price`
* name:string
* description:string
* amount:integer (cents)
* currency:string ("usd")
* metered:boolean
* unit_label:string
* charge_per_unit:boolean
* trial_period_days:integer
* hidden:boolean

It probably needs a JSON column for handling Stripe, Braintree, Paddle, etc, IDs, Stripe Tax, and any other details.

Alternatively, you could add `processor` and `processor_id` and duplicate this for each payment processor.
